File: sdk/typescript/src/namespaces/mcp.ts
/**
 * MCP Namespace API
 *
 * Provides access to MCP tool discovery routes.
 */

interface MCPClientInterface {
  request<T = unknown>(
    method: string,
    path: string,
    options?: { params?: Record<string, unknown>; body?: unknown; json?: Record<string, unknown> }
  ): Promise<T>;
}

export class MCPAPI {
  constructor(private client: MCPClientInterface) {}

  async listTools(category?: string): Promise<Record<string, unknown>> {
    const params = category ? { category } : undefined;
    return this.client.request('GET', '/api/v1/mcp/tools', { params });
  }

  async getTool(name: string): Promise<Record<string, unknown>> {
    return this.client.request('GET', `/api/v1/mcp/tools/${encodeURIComponent(name)}`);
  }
}

File: sdk/python/aragora_sdk/namespaces/mcp.py
"""
MCP namespace API.

Provides SDK access to Aragora's MCP tool discovery endpoints.
"""

from __future__ import annotations

from typing import TYPE_CHECKING, Any

if TYPE_CHECKING:
    from ..client import AragoraAsyncClient, AragoraClient


class MCPAPI:
    """Synchronous MCP tool discovery API."""

    def __init__(self, client: AragoraClient) -> None:
        self._client = client

    def list_tools(self, category: str | None = None) -> dict[str, Any]:
        """List MCP tools, optionally filtered by category prefix."""
        params: dict[str, Any] = {}
        if category:
            params["category"] = category
        return self._client.request("GET", "/api/v1/mcp/tools", params=params or None)

    def get_tool(self, name: str) -> dict[str, Any]:
        """Get one MCP tool by name."""
        return self._client.request("GET", f"/api/v1/mcp/tools/{name}")


class AsyncMCPAPI:
    """Asynchronous MCP tool discovery API."""

    def __init__(self, client: AragoraAsyncClient) -> None:
        self._client = client

    async def list_tools(self, category: str | None = None) -> dict[str, Any]:
        """List MCP tools, optionally filtered by category prefix."""
        params: dict[str, Any] = {}
        if category:
            params["category"] = category
        return await self._client.request("GET", "/api/v1/mcp/tools", params=params or None)

    async def get_tool(self, name: str) -> dict[str, Any]:
        """Get one MCP tool by name."""
        return await self._client.request("GET", f"/api/v1/mcp/tools/{name}")
